### PR TITLE
Go to Implementation improvements

### PR DESCRIPTION
**Pull request by jasonmalinowski**
This fixes invoking Go to Implementation on non-sealed classes and interface methods that have virtual implementations. It also finally offers a public API for finding derived types of a type.

Reviewing change-at-a-time might make this change a bit easier to stomach.

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
@dotnet/roslyn-ide, I can haz code review?

**Comment by balajikris** [comment]
LGTM.

**Comment by CyrusNajmabadi** [comment]
:+1:

File: src/EditorFeatures/Test2/GoToImplementation/GoToImplementationTests.vb [hunk 1]
```diff
@@ -50,6 +50,96 @@ class [|$$C|] { }
         End Function
 
         <WpfFact, Trait(Traits.Feature, Traits.Features.GoToImplementation)>
+        Public Async Function TestWithAbstractClass() As Task
+            ' Since the class is abstract, it cannot be an implementation of itself. Compare to TestWithSingleClass
+            ' above (where we count a class as an implementation of itself) or
```
Comment (rchande) [inline src/EditorFeatures/Test2/GoToImplementation/GoToImplementationTests.vb]: Or what? Such suspense!

**Comment by rchande** [comment]
:shipit:

**Comment by jasonmalinowski** [comment]
Did an updated push with some changes:

1. When computing overrides, I now use a HashSet since there are a few cases where you might get duplicated methods. Tests were added.
2. Resolved @rchande's suspense with the rest of the comment.

No interesting changes, so I'll assume existing signoffs still count.

**Comment by jasonmalinowski** [comment]
Only failures are tst queues.

**Pull request merged**